openwisp_notifications/handlers.py

```diff
@@ -203,25 +203,61 @@ def send_email_notification(sender, instance, created, **kwargs):
     else:
         target_url = None
     if target_url:
-        description += _('\n\nFor more information see %(target_url)s.') % {
-            'target_url': target_url
-        }
-
-    send_email(
-        subject,
-        description,
-        instance.message,
-        recipients=[instance.recipient.email],
-        extra_context={
-            'call_to_action_url': target_url,
-            'call_to_action_text': _('Find out more'),
-        },
-    )
+        description += _('\n\nFor more information see %(target_url)s.') % {'target_url': target_url}
 
-    # flag as emailed
-    instance.emailed = True
-    # bulk_update is used to prevent emitting post_save signal
-    Notification.objects.bulk_update([instance], ['emailed'])
+    return True, subject, description, target_url
+
+
+@receiver(post_save, sender=Notification, dispatch_uid='send_email_notification')
+def send_email_notification(sender, instance, created, **kwargs):
+    # Abort if a new notification is not created
+    if not created:
+        return
+
+    can_send, subject, description, target_url = can_send_email(instance)
+
+    if not can_send:
+        return
+
+    recipient_id = instance.recipient.id 
+    cache_key = f'email_batch_{recipient_id}'
+    cache_data = cache.get(cache_key, {'last_email_sent_time': None, 'batch_scheduled': False})
+    is_send_email = False
+
+    if cache_data['last_email_sent_time']:
+        if (timezone.now() - cache_data['last_email_sent_time']).seconds < EMAIL_BATCH_INTERVAL:
+            if not cache_data['batch_scheduled']:
```

Comment:
The fact that the `cache_data['last_email_sent_time']` exists should be enough to tell there's an scheduled task for the batch sending. 

We would need some comments here to explain all of this.